### PR TITLE
Use external provider_id when enumerating #9465

### PR DESCRIPTION
**Pull request by phebal**

### CODE REVIEW

**Comment by phebal** [comment]
This has been tested in umass-staging.

File: lib/ncs_navigator/core/warehouse/ineligible_batch_enumerator.rb [hunk 1]
```diff
@@ -11,7 +11,11 @@ class IneligibleBatchEnumerator
 
     produce_records(
       :ineligible_batches,
-      :query => "SELECT * FROM ineligible_batches"
+      :query => <<-SQL
+        SELECT p.provider_id provider_public_id, ib.*
```
Comment (hannahwhy) [inline lib/ncs_navigator/core/warehouse/ineligible_batch_enumerator.rb]: An `AS` would be useful here for clarity, but it's not really that big of a deal.  (The first time I looked at this, I thought a comma was missing.)

**Comment by hannahwhy** [comment]
Has @anassiri gotten back to you on whether this addresses the transform errors she saw?  (I don't see anything additional in the relevant hosted data ticket.)

**Comment by anassiri** [comment]
Test ETL process is running on Uofl data, should complete in around 4 hours.

**Comment by phebal** [comment]
This is only meant to address a bug in MDES record extraction for PBS batched ineligible women.  It's unlikely to solve any other data issues.

**Comment by hannahwhy** [comment]
@anassiri has told me that this addresses hosted data issue #9465; merging.

**Pull request merged**